### ISSUE TITLE
[next] exports field browser conditional ordering

### ISSUE THREAD

**Issue by yyx990803** [lukeed/httpie#36]
Ran into this when a user is trying to use `httpie@next` with Vite 2.

Currently the exports field has the following conditionals:

```json
  "exports": {
    ".": {
      "import": "./node/index.mjs",
      "require": "./node/index.js",
      "browser": "./xhr/index.mjs",
      "worker": "./fetch/index.mjs"
    }
  }
```

From the bundler's perspective, it will resolve the `import` conditional immediately because that matches the ESM environment. However, the file actually points to the node version.

According to the [docs](https://nodejs.org/api/packages.html#packages_conditional_exports):

> The general rule is that conditions should be from most specific to least specific in object order.

So in this case, `browser` should be listed **before** `import` in order for browser-targeting bundlers to prioritize it. It is ok to list it first because Node.js will ignore it. `worker` should probably be listed before `import` and `require` as well.

**Comment by lukeed** [comment]
Thank you! This was a TIL for me at the end of last week. I did a deep dive/revisit after that Twitter thread (you might be interested in what became of that). The next step was to revisit my stuff and make sure it complied with this aspect specifically.

Errands now, will fix in a bit

**Comment by lukeed** [comment]
Released `v2.0.0-next.13`  with `worker > browser > import > require`

Thanks again :)
